verifier/src/main/kotlin/tech/figure/classification/asset/verifier/config/EventStreamProvider.kt

```diff
@@ -1,16 +1,15 @@
 package tech.figure.classification.asset.verifier.config
 
-import io.provenance.eventstream.stream.clients.BlockData
 import tech.figure.classification.asset.verifier.provenance.AssetClassificationEvent
 
 interface EventStreamProvider {
     suspend fun currentHeight(): Long?
     suspend fun startProcessingFromHeight(
         height: Long? = null,
-        onBlock: (suspend (block: BlockData) -> Unit),
-        handleEvent: (suspend (event: AssetClassificationEvent) -> Unit),
+        onBlock: (suspend (blockHeight: Long) -> Unit),
+        onEvent: (suspend (event: AssetClassificationEvent) -> Unit),
         onError: (suspend (throwable: Throwable) -> Unit),
-        onCompletion: (suspend (throwable: Throwable?) -> Unit),
-        onNetAdapterShutdownFailure: (suspend (throwable: Throwable) -> Unit)
+        onCompletion: (suspend (throwable: Throwable?) -> Unit)
     )
+    suspend fun <T> toAssetClassificationEvent(data: T): List<AssetClassificationEvent>
```

Comment:
I apologize: I kinda went in a ton of directions with my comments lol.  I think this can actually be removed.  Each class that inherits from `EventStreamProvider` will just need its own way of converting its block data to an `AssetClassificationEvent`.  Having this as a generic function is actually detrimental, which can be noted when looking at your implementations.  I recommend just outright removing this, and converting each of those functions into private class functions in the default + block api providers.  

Then you can do away with the `is Block.X` checks in each of those functions, but keep them as a standard for emitting values to the `onEvent` block you have defined here.  Sorry again for the confusion